airflow/operators/python.py

```diff
@@ -377,9 +377,20 @@ def execute_callable(self):
                 render_template_as_native_obj=self.dag.render_template_as_native_obj,
             )
 
+            # find python executable folder
+            candidates = [os.path.join(tmp_dir, 'bin'), os.path.join(tmp_dir, 'scripts')]
+            python_folder = None
+            for candidate in candidates:
+                if os.path.isdir(candidate):
+                    python_folder = candidate
+                    break
+
+            if python_folder is None:
+                raise AirflowException(f'Unable to find python executable in "{tempdir}"')
+
             execute_in_subprocess(
                 cmd=[
-                    f'{tmp_dir}/bin/python',
+                    os.path.join(python_folder, 'python'),
```

Comment:
I think this can be simplified with `shutil.which`. Instead of trying to find a parent directory to `os.path.join` later:

```python
candidate_dirs = [os.path.join(tmp_dir, "bin"), os.path.join(tmp_dir, "Scripts")]
python_executable = shutil.which("python", path=os.pathsep.join(candidate_dirs))
if not python_executable:
    raise AirflowException(...)
execute_in_subprocess(cmd=[python_executable, ...])
```

Also, I feel the `AirflowException` should show `tmp_dir` instead. Showing `tempdir` can be misleading because the Python executable isn’t directly in the temp directory, but subdirectory `tmp_dir`.

Reply:
Very nice, I didn't know about this function. Updated in the latest commit.